downloads/openml_expdb.sql

```diff
@@ -148,6 +147,20 @@ CREATE TABLE `dataset_tag` (
 -- --------------------------------------------------------
 
 
+--
+-- Table structure for table `dataset_description`
+--
+
+CREATE TABLE `dataset_description` (
+  `did` int(10) UNSIGNED NOT NULL,
+  `version` varchar(64) COLLATE utf8_unicode_ci NOT NULL,
```

Comment:
probably better to make this a date time, instead of a integer?

Reply:
I used version numbers similar to the way it is done for dataset versioning. Do you think using an integer may cause problems if there are concurrent "Edits"?

Reply:
I think that `did` is the primary key of dataset, and `name`, `version` is there an alternative key, that gives users the option to specify dataset versions in a non-numeric way. 

I do not think we need that for the description version. `did` maps as foreign key to the dataset table, and version can just increment numerically (or according to some other logical scheme, such as a date).

Reply:
Concurrent edits are indeed a problem, with e.g., knowing which edit was the most recent and which you want to present to the user. How do you want to solve that?

(Separate question. I don't think that having a varchar data field solves the issue. )

Reply:
Could we just use datetime with a resolution of milli or microseconds and add it to a column.
I think concurrent updates would still differ in milli seconds if not seconds? Something like:
"2012-07-08 11:14:15.889342"

Reply:
I mean the problem is when two persons make an adjustment based on text A. Person 1 will maybe make a big update, whereas person 2 will improve some typos. If both are based upon version A, then the edits of person 1 will be stowed away in the database. But maybe that's something we have to accept.

I wouldn't be to worried about the database overwriting a certain entry, that could be easily avoided by relying on mysqls internal procedures.

Reply:
Yes, may be we could accept this behavior as it may be rare as well.

Reply:
"version can just increment numerically (or according to some other logical scheme, such as a date"

So we are incrementing the version numerically currently. I think we could go ahead with this, unless there is another reason to switch to date. We could accept that in concurrent edits there is a possibility that the database may reject a change and pick another(as simultaneous edits may be a rare occurrence)

Reply:
@janvanrijn I will create an API list_description_version(did, N) which can list all  or latest N versions of a description. This way it is possible to get the history and restore to an earlier version if something goes wrong.